xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/main/java/org/xwiki/mentions/internal/jmx/JMXMentionsMBean.java

```diff
@@ -37,4 +37,16 @@
      * Clear the mentions analysis tasks queue.
      */
     void clearQueue();
+
+    /**
+     * Update the number of thread consuming the user mentions analysis tasks.
+     *
+     * @param number the new number of threads.
+     */
+    void updateThreadNumber(int number);
```

Comment:
Well that's an interesting approach to allow changing dynamically the number of threads, but that's not really what I had in mind :) was more thinking about a config in xwiki.properties to specify the number of threads to use as startup, as we have for notifications for example.

Reply:
I'd say that we can keep your implem but we should still have the default number of threads computed from a property.

Reply:
mentions.poolSize property added. I've kept the dynamic jmx configuration for now.